test/java/seedu/addressbook/commands/AddCommandTest.java

```diff
@@ -0,0 +1,156 @@
+                    Address.EXAMPLE, false, NO_TAGS);
+        }
+    }
+
+    @Test
+    public void addCommand_invalidPhone_throws() {
+        final String[] invalidNumbers = { "", " ", "1234-5678", "[]\\[;]", "abc", "a123", "+651234" };
+        for (String number : invalidNumbers) {
+            assertConstructingInvalidAddCmdThrows(Name.EXAMPLE, number, false, Email.EXAMPLE, false,
+                    Address.EXAMPLE, false, NO_TAGS);
+        }
+    }
+
+    @Test
+    public void addCommand_invalidEmail_throws() {
+        final String[] invalidEmails = { "", " ", "def.com", "@", "@def", "@def.com", "abc@",
+                "@invalid@email", "invalid@email!", "!invalid@email" };
+        for (String email : invalidEmails) {
+            assertConstructingInvalidAddCmdThrows(Name.EXAMPLE, Phone.EXAMPLE, false, email, false,
+                    Address.EXAMPLE, false, NO_TAGS);
+        }
+    }
+
+    @Test
+    public void addCommand_invalidAddress_throws() {
+        final String[] invalidAddresses = { "", " " };
+        for (String address : invalidAddresses) {
+            assertConstructingInvalidAddCmdThrows(Name.EXAMPLE, Phone.EXAMPLE, false, Email.EXAMPLE, false,
+                    address, false, NO_TAGS);
+        }
+    }
+
+    @Test
+    public void addCommand_invalidTags_throws() {
+        final String[][] invalidTags = { { "" }, { " " }, { "'" }, { "[]\\[;]" }, { "validTag", "" },
+                { "", " " } };
+        for (String[] tags : invalidTags) {
+            Set<String> tagsToAdd = new HashSet<>(Arrays.asList(tags));
+            assertConstructingInvalidAddCmdThrows(Name.EXAMPLE, Phone.EXAMPLE, false, Email.EXAMPLE, false,
+                    Address.EXAMPLE, false, tagsToAdd);
+        }
+    }
+
+    /**
+     * Asserts that attempting to construct an add command with the supplied
+     * invalid data throws an IllegalValueException
+     */
+    private void assertConstructingInvalidAddCmdThrows(String name, String phone, boolean isPhonePrivate,
+            String email, boolean isEmailPrivate, String address, boolean isAddressPrivate,
+            Set<String> tags) {
+        try {
+            new AddCommand(name, phone, isPhonePrivate, email, isEmailPrivate, address, isAddressPrivate,
+                    tags);
+        } catch (IllegalValueException e) {
+            return;
+        }
+        String error = String.format(
+                "An add command was successfully constructed with invalid input: %s %s %s %s %s %s %s %s",
+                name, phone, isPhonePrivate, email, isEmailPrivate, address, isAddressPrivate, tags);
+        fail(error);
+    }
+
+    @Test
+    public void addCommand_validData_correctlyConstructed() throws Exception {
+        AddCommand cmd = new AddCommand(Name.EXAMPLE, Phone.EXAMPLE, true, Email.EXAMPLE, false,
+                Address.EXAMPLE, true, NO_TAGS); // should never throw
```

Comment:
Comment can be removed. Not throwing an exception is the normal expectation that goes without saying.